### DISCUSSION TITLE
Chirality in Compounds With Implicit Hydrogen Stereochemistry Not Preserved

### DISCUSSION THREAD

**Discussion by BehrgenSmith** [rdkit/rdkit#7517] [Q&A]
This was done in RDKit 2022.09.1

```
from rdkit.Chem import AllChem as Chem
smile = 'Brc1ccc(C[N+]C[C@]2OCCC2)cc1'
'mol = Chem.MolFromSmiles(smile)
Chem.MolToSmiles(mol, isomericSmiles=True)
```
`Output : 'Brc1ccc(C[N+]C[C]2CCCO2)cc1'`

As you can see the chirality is not preserved, a similar case can be seen for F[C@@](N)C and F[C@@H](N)C.

If the chirality inducing hydrogen is specified it works:

```
smile = 'Brc1ccc(C[N+]C[C@H]2OCCC2)cc1'
mol = Chem.MolFromSmiles(smile)
Chem.MolToSmiles(mol, isomericSmiles=True)
```
`Output : 'Brc1ccc(C[N+]C[C@@H]2CCCO2)cc1'`

Our established workflows that created large internal databases at our organization do not record explicit hydrogens. We believe that it should be possible to read the chirality without the explicit hydrogen. Is this a bug in RDKit? Is there any other flags or protocols that could be used to preserve the chirality specified in the intial SMILES string without the explicit hydrogen?

**Comment by bp-kelley** [comment]
When in a bracket, hydrogens have to be explicit, they aren't determined implicitly

```
Brc1ccc(C[N+]C[C@H]2OCCC2)cc1
```

Is not the same as

```
Brc1ccc(C[N+]C[C@]2OCCC2)cc1
```

You can investigate this using the "Debug" function

```
>>> Chem.MolFromSmiles("[C@]").Debug()
Atoms:
	0 6 C chg: 0  deg: 0 exp: 0 imp: 0 hyb: SP3 rad: 4
Bonds:
>>> Chem.MolFromSmiles("[C@H]").Debug()
Atoms:
	0 6 C chg: 0  deg: 0 exp: 0 imp: 1 hyb: SP3 rad: 3
Bonds:
>>> Chem.MolFromSmiles("C").Debug()
Atoms:
	0 6 C chg: 0  deg: 0 exp: 0 imp: 4 hyb: SP3
Bonds:
```

**Reply by BehrgenSmith** [reply]
Maybe this is naive, but in Brc1ccc(C[N+]C[C@]2OCCC2)cc1, isn't the carbon missing a substituent and therefore a technically invalid SMILES?
